### PR TITLE
Include path in the search index with include_path

### LINKED ISSUE #1481: Include path in the search index
Would be useful on my wiki as pages don't have a title, instead the path serves as the replacement for a title. I'd be willing to PR if you agree to add this feature (in form of a `search/include_path` switch).

### PR DESCRIPTION
**Pull request by timakro**
This way URLs can be searched conveniently on wiki- / knowledge base- / docs-style sites. E.g. the path `/documentation/getting-started/overview/` at https://getzola.org would be matched by queries like "documentation" or "getting started".

Closes #1481.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?

### CODE REVIEW

File: components/search/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ authors = ["Vincent Prouillet <prouillet.vincent@gmail.com>"]
 edition = "2018"
 
 [dependencies]
-elasticlunr-rs = {version = "2", default-features = false, features = ["da", "de", "du", "es", "fi", "fr", "it", "pt", "ro", "ru", "sv", "tr"] }
+elasticlunr-rs = {version = "2.3.12", default-features = false, features = ["da", "de", "du", "es", "fi", "fr", "it", "pt", "ro", "ru", "sv", "tr"] }
```
Comment (Keats) [inline components/search/Cargo.toml]: you don't need to change that, Cargo will automatically use the latest version in this case

File: components/search/src/lib.rs [hunk 2]
```diff
@@ -36,17 +38,57 @@ fn build_fields(search_config: &Search) -> Vec<String> {
...
+
+    if search_config.include_path {
+        tokenizers.push(path_tokenizer);
+    }
+
+    if search_config.include_content {
+        tokenizers.push(text_tokenizer);
+    }
+
+    tokenizers
```
Comment (Keats) [inline components/search/src/lib.rs:84]: You can end up with the same tokenizer multiple times, is that not an issue?

**Review by Keats — CHANGES_REQUESTED** [review]

**Comment by Keats** [comment]
>  the path /documentation/getting-started/overview/ at https://getzola.org would be matched by queries like "documentation" or "getting started".

Does that mean that the index will be essentially filled with "useless" data in this case? Every doc page starts with documentation

File: components/search/src/lib.rs (hunk 2 above)
Comment (timakro) [reply components/search/src/lib.rs:84]: The same tokenizer, that is the default tokenizer, should be used for all fields except the path field which gets it's own tokenizer that splits on `/`. Those are just function pointers so having two pointers point to the same function is not an issue.

**Comment by timakro** [comment]
On my wiki I have a bunch of pages in the `hack` section. All of them are indexed under "hack":

image

So when you search for "hack" you find those pages. To me this seems like the behavior you'd expect.

You probably only want to use path search when the path of the pages is exposed to your users in some way. For example if you show the path of the pages in the search results and highlight the matches. Or if you show the path on top of the page e.g. in breadcrumbs style. Then matching all docs pages when searching for "documentation" makes sense.

image

**Review by Keats — APPROVED** [review]

**Comment by timakro** [comment]
Thanks for taking the time to review my code. And thanks for creating this terrific piece of software! :)

**Comment by Keats** [comment]
You're welcome!

**Pull request merged**